patches/tModLoader/Terraria/Player.cs.patch

```diff
@@ -7964,8 +7974,21 @@
  			fileIO.Write(newPlayer.builderAccStatus[num9]);
  		}
  
-@@ -42272,6 +_,10 @@
+@@ -42270,12 +_,27 @@
+ 		if (Main.rand == null)
+ 			Main.rand = new UnifiedRandom((int)DateTime.Now.Ticks);
  
++		var plrData = FileUtilities.ReadAllBytes(playerPath, cloudSave);
++		// Read data in advance instead of halfway
++		// What if there is an exception in PlayerIO.TryLoadData? It is not handled the same as before
++		if (!PlayerIO.TryLoadData(playerFileData.Path, playerFileData.IsCloudSave, out var tplrData))
++			tplrData = null;
++
++		return LoadPlayerFromStream(playerFileData, plrData, tplrData);
++	}
++
++	public static PlayerFileData LoadPlayerFromStream(PlayerFileData playerFileData, byte[] plrData,TagCompound tplrData)
```

Comment:
As you noted, this messes with the current exception handling. I'm surprised you need `LoadPlayerFromStream` and that `Deserialize` isn't sufficient for you. There's not much else in `LoadPlayerFromStream` other than error handling and 3 lines of post-processing.

If you need this method, then the options are:
- Pass `byte[] tplrData` (preferred)
- Pass `Func<PlayerFileData, TagCompound> getTplrData`

Reply:
Serialize and Deserialize are only applicable to player data, while PLR files will have encrypted headers and metadata.

I separated LoadPlayerFromStream instead of using Deserialize,hoping to use PLR binary as a arg to obtain the player without processing the header (ENCRYPTION_KEY is private and requires reflection).